### PR TITLE
Documentation enhancements and changes in repo.R

### PR DESCRIPTION
**Pull request by thereseanders**
Implements a number of changes in repo.R and related files:
1. Adds vectorized version of `check_repo` that implements check for renamed repositories.
2. uses `github_api_get_collaborators()` within `get_collaborators`
3. Deprecates `check_repos` and `get_repo_collaborators`
4. Changes name of function `github_api_repo_exists` to `github_api_get_repo`("exists" suggests a logical output, but the output is a nested list object with repository information)
5. Adds documentation to all exported functions in repo.R

**Edit 6/8/19**
6. Added more `github_api_*` functions and replaced `gh()`calls in `repo.R` with them.

### CODE REVIEW

**Comment by thereseanders** [comment]
Added more `github_api_*` functions and replaced `gh()`calls in `repo.R` with them.

**Comment by thereseanders** [comment]
Added more `github_api_*` functions and replaced `gh()`calls in `repo.R` with them.

File: R/repo.R [hunk 1]
```diff
@@ -1,83 +1,227 @@
-github_api_get_repo_collaborators = function(repo) {
-  gh(
+github_api_get_collaborators = function(repo) {
```
Comment (mine-cetinkaya-rundel) [inline R/repo.R]: We're making this singular, right?

File: R/repo.R [hunk 2]
```diff
@@ -1,83 +1,227 @@
...
+  safe_gh(
     "GET /repos/:owner/:repo/collaborators",
     owner = get_repo_owner(repo), repo = get_repo_name(repo),
     .token=get_github_token(), .limit=get_github_api_limit()
```
[the remark this answers is not in the document]
Comment (thereseanders) [reply R/repo.R]: Added spaced before and after + throughout document in #baf70adc1499061ab6149d2b2624bd684b3fcebd

File: R/repo.R [hunk 3]
```diff
@@ -1,83 +1,227 @@
...
   )
 }
 
 
-#' @export
+#' List repository collaborators
+#'
+#' \code{get_repo_collaborators} returns collaborator usernames.
```
[the remark this answers is not in the document]
Comment (thereseanders) [reply R/repo.R]: Implemented through commit https://github.com/thereseanders/ghclass/commit/f1ad7f6eeb095f836e1e9022e9ce410ab9d08d28.

File: R/repo.R [hunk 4]
```diff
@@ -1,83 +1,227 @@
...
+    repo_exists = ifelse(repo_exists & current_name != repo_name, F, repo_exists)
+  }
+
+  # Messaging
+  purrr::walk2(current_name,
+               repo_name,
+               function(current_name, repo_name)
 
+                 if(current_name != repo_name & !is.na(current_name))
+                   message(paste("Repository", repo_name, "was previously renamed to", current_name)))
```
[the remark this answers is not in the document]
Comment (thereseanders) [reply R/repo.R]: Implemented per https://github.com/thereseanders/ghclass/commit/2770d59c6e4249fb6d1ec67e3f980f0bc412ead1 and https://github.com/thereseanders/ghclass/commit/25b726cb75f14ec02c313613464ce52f73fa5212. Note that the `repo_old_name` and `repo_new_name` parameters are internally created and not user supplied, thus not documented.

File: R/repo.R (hunk 1 above)
Comment (thereseanders) [reply R/repo.R]: Implemented per https://github.com/thereseanders/ghclass/commit/586dc52af6015e9749d811eadf5afd71013d61af.

**Comment by thereseanders** [comment]
- Commit https://github.com/thereseanders/ghclass/commit/586dc52af6015e9749d811eadf5afd71013d61af changes the `org` parameter to `owner` throughout. 
- Commit https://github.com/thereseanders/ghclass/commit/c5dd144974e0938422580b18a06ef582d1de00dc re-builds all documentation and pkgdown site. Now deprecated functions are no longer included in the documentation/site.

**Pull request merged**